src/Symfony/Component/TypeInfo/Type/ExplicitStringType.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo\Type;
+
+use Symfony\Component\TypeInfo\TypeIdentifier;
+
+/**
+ * Explicit string type.
+ *
+ * @author Martin Rademacher <mano@radebatz.net>
+ *
+ * @extends BuiltinType<TypeIdentifier::STRING>
+ */
+class ExplicitStringType extends BuiltinType
+{
+    public function __construct(private string $explicitType)
```

Comment:
Indeed, I do think we need an enum here:
```suggestion
    public function __construct(private StringConstraint $constraint)
```
And then maybe `ConstrainedStringType` is better?

By doing that, we'll be able to implement the `accepts` method as well.